src/App.js

```diff
@@ -1,14 +1,37 @@
-import { React, useState } from "react";
+import { React, useState, useEffect } from "react";
 
 import Counter from "./Counter";
 import CounterDisplay from "./CounterDisplay";
 
-function App() {
+// I create custom hooks when I want to be able to reuse them for multiple components
+// It isn't necessary here, but I'm including it for the sake of the demo
+const useCustomState = () => {
   const [clickCount, setClickCount] = useState(0);
-  console.log("clickCount:", clickCount);
+
+  // This will only run when the variable clickCount changes
+  useEffect(() => {
+    console.log("the click count changed!");
+    console.log(clickCount);
+  }, [clickCount]);
```

Comment:
This is the dependency array - what's in this array `[clickCount]` is what React is going to listen to. When `clickCount` changes, the code in this `useEffect` block will run.